### ISSUE TITLE
Tons of warnings on use of TransfoXLModel. masked_fill_ input dtype torch.uint8 should be changed to torch.bool

### ISSUE THREAD

**Issue by CrafterKolyan** [huggingface/pytorch-transformers#1119]
## 🕑 Usage of deprecated behaviour

Using example from documentation web page: https://huggingface.co/pytorch-transformers/model_doc/transformerxl.html#pytorch_transformers.TransfoXLModel
```
import torch
from pytorch_transformers import *

tokenizer = TransfoXLTokenizer.from_pretrained('transfo-xl-wt103')
model = TransfoXLModel.from_pretrained('transfo-xl-wt103')
input_ids = torch.tensor(tokenizer.encode("Hello, my dog is cute")).unsqueeze(0)
outputs = model(input_ids)
last_hidden_states, mems = outputs[:2]
```

Get tons of same warnings:
> /pytorch/aten/src/ATen/native/LegacyDefinitions.cpp:14: UserWarning: masked_fill_ received a mask with dtype torch.uint8, this behavior is now deprecated,please use a mask with dtype torch.bool instead.
![image](https://user-images.githubusercontent.com/9883873/63766644-4b180c80-c8d4-11e9-890c-9da640b5c0e8.png)

Created #1120 to fix it.
